### PR TITLE
Aggressive cows

### LINKED ISSUE #1903: Aggressive cows in java
Farmer John has built a new long barn, with N (2 <= N <= 100,000) stalls. The stalls are located along a straight line at positions x1,...,xN (0 <= xi <= 1,000,000,000).

His C (2 <= C <= N) cows don't like this barn layout and become aggressive towards each other once put into a stall. To prevent the cows from hurting each other, FJ wants to assign the cows to the stalls, such that the minimum distance between any two of them is as large as possible. What is the largest minimum distance?
Input

Line 1: Two space-separated integers: N and C
Lines 2..N+1: Line i+1 contains an integer stall location, xi
Output
For each test case output one integer: the largest minimum distance.
Example
Input:

5 3
1
2
8
4
9
Output:

3
Output details:

FJ can put his 3 cows in the stalls at positions 1, 4 and 8,
resulting in a minimum distance of 3.

### PR DESCRIPTION
**Pull request by j-shreya**
Fixes #1903
 
#### Checklist:
- [X] 4 space indentation.
- [X] Coding conventions are followed.
- [X] Input is taken dynamically.
- [X] Sample Input / Output is added at the end of file.
- [X] Logic Documentation (Comments).
- [X] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- Java
- 

#### Files Added:
- Aggressive_Cows.java
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by SakshiShreya** [comment]
Are you working on this issue? I can't see any files.

File: Aggressive_Cows/Aggressive_Cows.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+public class Aggressive_Cows {
+
+    public static void main(String[] args) {
+
+        // in this problem we have to maximise the minimum
+        // distance between the cows
+        // since minimum distance is being linearly checked
+        // we apply binary search on
+        // minimum
+        // distance
```
Comment (asha15) [inline Aggressive_Cows/Aggressive_Cows.java]: make this as a multi line command

File: Aggressive_Cows/Aggressive_Cows.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        while (lo <= hi) {
+
+            int mid = (lo + hi) / 2;
+
+            if (isItPossible(nos, noc, arr, mid)) {
+
+                finalAns = mid;
+                lo = mid + 1;
+
+            } else {
```
Comment (asha15) [inline Aggressive_Cows/Aggressive_Cows.java]: remove the space before and after `else`

File: Aggressive_Cows/Aggressive_Cows.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                hi = mid - 1;
+            }
+
```
Comment (asha15) [inline Aggressive_Cows/Aggressive_Cows.java]: remove this line

File: Aggressive_Cows/Aggressive_Cows.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                lastCowPos = arr[i];
+
+                if (cowsPlaced == noc) {
+                    return true;
+                }
+            }
+
+        }
+
+        return false;
```
Comment (asha15) [inline Aggressive_Cows/Aggressive_Cows.java]: Correct the indentation 
remove the extra lines

File: Aggressive_Cows/Aggressive_Cows.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    }
+
```
Comment (asha15) [inline Aggressive_Cows/Aggressive_Cows.java]: remove the extra lines

**Comment by j-shreya** [comment]
Requested changes made . Pls check @asha15 , @SakshiShreya

**Pull request merged**